pyproject.toml

```diff
@@ -0,0 +1,72 @@
+[build-system]
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "soundata"
+dynamic = ["version"]
+description = "Python library for loading and working with sound datasets."
+readme = "README.md"
+license = {file = "LICENSE"}
+requires-python = ">=3.7"
+authors = [
+    {name = "Magdalena Fuentes", email = "mfuentes@nyu.edu"},
+]
+maintainers = [
+    {name = "Magdalena Fuentes", email = "mfuentes@nyu.edu"},
+]
+keywords = ["sound", "dataset loader", "audio"]
+classifiers=[
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Science/Research",
+    "Topic :: Multimedia :: Sound/Audio :: Analysis",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10"
+]
+dependencies = [
+    "librosa>=0.10.0",
+    "numpy>=1.21.6",
+    "pandas>=1.3.5",
+    "tqdm>=4.65.0",
+    "jams>=0.3.4",
+    "py7zr>=0.16.0",
+]
+
+[project.optional-dependencies]
+tests = [
+    "requests>=2.31.0",
+    "pytest>=7.2.0",
+    "pytest-cov>=4.1.0",
+    "pytest-pep8>=1.0.6",
+    "pytest-mock>=3.10.0",
+    "pytest-localserver>=0.7.1",
+    "pytest-xdist>=2.3.0",
+    "future>=0.18.3",
+    "coveralls>=3.3.1",
+    "black>=23.3.0",
+    "flake8>=5.0.4",
+    "mypy>=0.982",
+]
+docs = [
```

Comment:
You are recommending to install the `docs` requirements via this file. I guess you can then remove `docs/requirements.txt` altogether?

Reply:
This file is currently used here: [.readthedocs.yml](https://github.com/soundata/soundata/blob/5e7aebcfb5b048bfdb2f8a91fe19b4c9a3b2738c/.readthedocs.yml#L26). Maybe we can remove the whole optional python installation instructions is this file since contributors will be already installing the required documentation dependencies through pyproject if they want to contribute (although we might need to document that further in the documentation)

Reply:
I've found this https://docs.readthedocs.io/en/stable/config-file/v2.html#packages so according to it we should be able to specify the requirements with .toml file and specify the readthedocs requirements with:
```yaml
python:
  install:
    - method: pip
      path: .
      extra_requirements:
        - docs
```
that will run `pip install .[docs]`

Reply:
Thanks, now we have a single source of truth for the documentation requirements.